### PR TITLE
Allow precision-recall to handle 2d predicted probabilities; docs updates

### LINKED ISSUE #2083: Add user guide entry for pipeline save/load methods
On [our pipelines user guide page](https://evalml.alteryx.com/en/stable/user_guide/pipelines.html) I'd say something like:
```python
import evalml
X, y = evalml.demos.load_breast_cancer()
class MyBinaryClassificationPipeline(BinaryClassificationPipeline):
    component_graph = ['Imputer', 'Random Forest Classifier']
pipeline = MyBinaryClassificationPipeline({})
pipeline.fit(X, y)
pipeline.save('~/pipeline.cloudpickle')

# then in another python process you can load the pipeline and use it
pipeline = evalml.pipelines.PipelineBase.load('~/pipeline.cloudpickle')
pipeline.predict(X_new)
```

### PR DESCRIPTION
**Pull request by dsherry**
Changes:
* Minor fix to precision-recall graph method: use same code as ROC method to process input data
* User guide updates: `train_pipelines`/`score_pipelines`, automl and pipeline save/load (fix #2083 ), move DT viz lower in model understanding guide. Update automl guide to use training-holdout split. 
* Add note to contributers guide about GitHub Actions permissions
* Fix typo in `PipelineBase` error.

### CODE REVIEW

File: docs/source/user_guide/automl.ipynb [hunk 1]
```diff
@@ -383,8 +453,12 @@
    "metadata": {},
    "outputs": [],
    "source": [
+    "# saving the entire automl search\n",
```
Comment (chukarsten) [inline docs/source/user_guide/automl.ipynb:456]: Isn't Angela's native pickle ready to go?

File: docs/source/user_guide/model_understanding.ipynb [hunk 2]
```diff
@@ -27,7 +27,7 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "First, let's train a pipeline on some data."
+    "First let's train a pipeline on some data."
```
Comment (chukarsten) [inline docs/source/user_guide/model_understanding.ipynb:30]: lol this is a nit, but I think the comma after the intro word is correct

File: docs/source/user_guide/model_understanding.ipynb [hunk 3]
```diff
@@ -375,6 +338,43 @@
     "graph_prediction_vs_actual(y_test, y_pred, outlier_threshold=50)\n"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now let's train a decision tree on some data."
```
Comment (chukarsten) [inline docs/source/user_guide/model_understanding.ipynb:345]: Begun, the Grammar Wars have.

**Review by chukarsten — APPROVED** [review]
An approval for this PR, I have given.  Ignore the grammar comments, you may.

image

File: docs/source/user_guide/automl.ipynb (hunk 1 above)
Comment (dsherry) [reply docs/source/user_guide/automl.ipynb:456]: Yes! Native pipeline pickles are good. Updating the docs with that info is tracked by #2158. And verifying that `AutoMLSearch` instances themselves are native-pickleable is tracked by #2174.

File: docs/source/user_guide/model_understanding.ipynb (hunk 2 above)
Comment (dsherry) [reply docs/source/user_guide/model_understanding.ipynb:30]: I am pretty sure both are acceptable. However, I don't always grammar real good. Will revert.

**Pull request merged**